docs/paper/dev/getting-started/paper-plugins.md

```diff
@@ -0,0 +1,116 @@
+able to define if a dependency is required during bootstrapping. This means that any bootstrapping
+logic for your dependencies will be run before yours is run. 
+```yml
+dependencies:
+  - name: DependencyName
+    required: true
+    bootstrap: true
+```
+
+This dependency is ``required``, meaning that if this dependency is not found your plugin will fail to load.
+The bootstrap field indicates that this dependency is required during bootstrapping, which currently does not serve a purpose.
+
+## What is it used for?
+Paper plugins lay down the framework for some future API.
+
+
+## Bootstrapper
+Paper plugins are able to identify their own bootstrapper by implementing ``io.papermc.paper.plugin.bootstrap.PluginBootstrap`` and adding
+the class of your implementation to the bootstrapper field in the ``paper-plugin.yml``.
+```java
+public class TestPluginBootstrap implements PluginBootstrap {
+
+    @Override
+    public void bootstrap(@NotNull PluginProviderContext context) {
+        
+    }
+
+    @Override
+    public @NotNull JavaPlugin createPlugin(@NotNull PluginProviderContext context) {
+        return new TestPlugin("My custom parameter");
+    }
+    
+}
+```
+A Bootstrapper allows you to also override how your plugin is initiated, allowing you to pass values into your plugin constructor.
+Currently, bootstrappers do not offer much new API, and are highly experimental. This may be subject to change once more API is introduced.
+
+## Loaders
+Paper plugins are able to identify their own plugin loader by implementing ``io.papermc.paper.plugin.loader.PluginLoader`` and adding
+the class of your implementation to the loader field in the ``paper-plugin.yml``.
+
+The goal of the plugin loader is the creation of an expected/dynamic environment for the plugin to load into. 
+This, as of right now, only applies to creating the expected classpath for the plugin, e.g. supplying external libraries to the plugin.
+```java
+public class TestPluginLoader implements PluginLoader {
+
+    @Override
+    public void classloader(@NotNull PluginClasspathBuilder classpathBuilder) {
+        classpathBuilder.addLibrary(new JarLibrary(Path.of("dependency.jar")));
+
+        MavenLibraryResolver resolver = new MavenLibraryResolver();
+        resolver.addDependency(new Dependency(new DefaultArtifact("com.example:example:version"), null));
+        resolver.addRepository(new RemoteRepository.Builder("paper", "default", "https://repo.papermc.io/repository/maven-public/").build());
+
+        classpathBuilder.addLibrary(resolver);
+    }
+}
+
+```
+Currently, you are able to add two different library types, ``JarLibrary``, and ``MavenLibraryResolver``.
+
+
+## Differences 
+
+### Bukkit Serialization System
+Paper plugins do not support the serialization system (``org.bukkit.configuration.serialization``) that bukkit uses. Your classes will not be able to be
+serialized, vise versa. It is highly recommended to not use this system with Paper Plugins.
```

Comment:
> Your classes will be unable to be serialized or de-serialized. As a result, please do not use this system with your Paper plugins.

I also lower cased `Plugins` here because it's not capitalized anywhere else.